### PR TITLE
Test examples in travis CI

### PR DESCRIPTION
**Pull request by neerajprad**
Adds `test_example` to ensure that our examples are not outdated due to changes in the code.

Other minor clean-ups:
 - Changes initialize_model to take in `*args, **kwargs`.
 - Moved examples to the outermost directory, so that we can have separate dependency requirements for examples just like notebooks.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,7 +14,7 @@ __pycache__/
 
 # data files
 numpyro/examples/.data
```
Comment (fehiepsi) [inline .gitignore]: nit: move this .data folder too

Comment (neerajprad) [reply .gitignore]: @fehiepsi - Note that all example utilities like `datasets.py` will still  be in `numpyro/examples` and the location of the `.data` will therefore not change. This is to make it easy for all examples to reuse these utilities (the `examples` folder isn't a python module), and also to share datasets.

Comment (fehiepsi) [reply .gitignore]: Ah, got it! Thank you for clarifying!

Comment (fehiepsi) [reply .gitignore]: Ah, got it! Thank you for clarifying!

**Comment by fehiepsi** [comment]
@neerajprad feel free to merge this when tests pass. It is looking like some library is missing for running covtype and vae.

**Pull request merged**